avm/res/elastic-san/elastic-san/tests/e2e/waf-aligned/main.test.bicep

```diff
@@ -0,0 +1,154 @@
+  params: {
+    virtualNetworkName: 'dep-${namePrefix}-vnet-${serviceShort}'
+    managedIdentityName: 'dep-${namePrefix}-msi-${serviceShort}'
+    // Adding base time to make the name unique as purge protection must be enabled (but may not be longer than 24 characters total)
+    keyVaultName: 'dep-${namePrefix}-kv-${serviceShort}-${substring(uniqueString(baseTime), 0, 3)}'
+    location: enforcedLocation
+  }
+}
+
+// Diagnostics
+// ===========
+module diagnosticDependencies '../../../../../../utilities/e2e-template-assets/templates/diagnostic.dependencies.bicep' = {
+  scope: resourceGroup
+  name: '${uniqueString(deployment().name, enforcedLocation)}-diagnosticDependencies'
+  params: {
+    storageAccountName: 'dep${namePrefix}diasa${serviceShort}01'
+    logAnalyticsWorkspaceName: 'dep-${namePrefix}-law-${serviceShort}'
+    eventHubNamespaceEventHubName: 'dep-${namePrefix}-evh-${serviceShort}'
+    eventHubNamespaceName: 'dep-${namePrefix}-evhns-${serviceShort}'
+    location: enforcedLocation
+  }
+}
+
+// General resources
+// =================
+resource resourceGroup 'Microsoft.Resources/resourceGroups@2021-04-01' = {
+  name: resourceGroupName
+  location: enforcedLocation
+}
+
+// ============== //
+// Test Execution //
+// ============== //
+
+var tags = {
+  Owner: 'Contoso'
+  CostCenter: '123-456-789'
+}
+
+@sys.batchSize(1)
+module testDeployment '../../../main.bicep' = [
+  for iteration in ['init', 'idem']: {
+    scope: resourceGroup
+    name: '${uniqueString(deployment().name, enforcedLocation)}-test-${serviceShort}-${iteration}'
+    params: {
+      name: '${namePrefix}${serviceShort}001'
+      sku: 'Premium_ZRS'
+      publicNetworkAccess: 'Disabled'
+      volumeGroups: [
+        {
+          name: 'vol-grp-01'
+          volumes: [
+            {
+              name: 'vol-grp-01-vol-01'
+              sizeGiB: 1
+            }
+          ]
+          managedIdentities: {
+            userAssignedResourceIds: [
+              nestedDependencies.outputs.managedIdentityResourceId
+            ]
+          }
+          customerManagedKey: {
+            keyName: nestedDependencies.outputs.keyVaultEncryptionKeyName
+            keyVaultResourceId: nestedDependencies.outputs.keyVaultResourceId
+            userAssignedIdentityResourceId: nestedDependencies.outputs.managedIdentityResourceId
+          }
+          privateEndpoints: [
+            {
+              subnetResourceId: nestedDependencies.outputs.subnetResourceId
+              privateDnsZoneGroup: {
+                privateDnsZoneGroupConfigs: [
+                  {
+                    privateDnsZoneResourceId: nestedDependencies.outputs.privateDNSZoneResourceId
+                  }
+                ]
+              }
+              tags: tags
+            }
+          ]
+        }
+      ]
+      tags: tags
+      lock: {
+        kind: 'CanNotDelete'
+        name: 'myCustomLockName'
+      }
```

Comment:
```suggestion
```

Not required for WAF-alignment

Reply:
Removed Lock from the WAF Test Case. Lock is tested only in the PE test case where the lock is set on PEPs.